src/sql/parts/notebook/notebookInput.ts

```diff
@@ -142,6 +143,7 @@ export class NotebookInput extends EditorInput {
 
 	constructor(private _title: string,
 		private resource: URI,
+		private _notebookEditorInput: UntitledEditorInput,
```

Comment:
In the QueryInput they manage to always have an input passed in, but for us it seems like it's only in the Untitled case. I'm worried (somewhat) about losing track and assuming this is non-null in a bunch of places. Should we consider always passing either the UntitledEditorInput or a full TextFileInput of some kind instead of trying to always cover both scenarios?

Reply:
UntitledEditorInput would be overwritten by FileEditorInput when we open an exisitng notebook. The only scenario this would become 'undfined' is when we open an existing notebook from "MainThreadNotebookDocumentsAndEditors" which I don't see a single use case. However, I will test it more and fix if there are any issuess